workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Worker;
+using Improbable.Worker.Core;
+using Unity.Entities;
+using UnityEngine;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.Core
+{
+    public class Worker : IDisposable
+    {
+        internal readonly Dictionary<EntityId, Entity> EntityMapping = new Dictionary<EntityId, Entity>();
+
+        public readonly Vector3 Origin;
+        public readonly string WorkerType;
+        public readonly string WorkerId;
+        public readonly World World;
+        public readonly Entity WorkerEntity;
+        public readonly Connection Connection;
+        public readonly ILogDispatcher LogDispatcher;
+
+        public delegate void WorkerEventHandler(Worker worker);
+        public static event WorkerEventHandler OnConnect;
+        public static event WorkerEventHandler OnDisconnect;
+
+        private Worker(ConnectionConfig config, ILogDispatcher logDispatcher, Vector3 origin)
+        {
+            Origin = origin;
+            WorkerId = string.IsNullOrEmpty(config.WorkerId)
+                ? $"{config.WorkerType}-{Guid.NewGuid()}"
+                : config.WorkerId;
+
+            Connection = ConnectionUtility.Connect(config, WorkerId);
+            LogDispatcher = logDispatcher;
+            WorkerType = config.WorkerType;
+            if (LogDispatcher is ForwardingDispatcher dispatcher)
+            {
+                dispatcher.SetConnection(Connection);
+            }
+
+            World = new World(WorkerId);
+            World.CreateManager<WorkerSystem>(this);
+            var entityManager = World.GetOrCreateManager<EntityManager>();
+            WorkerEntity = entityManager.CreateEntity(typeof(OnConnected), typeof(WorkerEntityTag));
+            OnConnect?.Invoke(this);
+        }
+
+        public static Worker TryGetWorker(World world)
+        {
+            var workerSystem = world.GetExistingManager<WorkerSystem>();
+            if (workerSystem == null)
+            {
+                throw new NullReferenceException("This world does not have a worker associated with it.");
+            }
+            return workerSystem.Worker;
+        }
+
+        public bool TryGetEntity(EntityId entityId, out Entity entity)
+        {
+            return EntityMapping.TryGetValue(entityId, out entity);
+        }
+
+        public static Worker Connect(ConnectionConfig config, ILogDispatcher logDispatcher, Vector3 origin)
+        {
+            return new Worker(config, logDispatcher, origin);
+        }
+
+        public static void Disconnect(Worker worker)
+        {
+            worker.Dispose();
+        }
+
+        public void Dispose()
```

Comment:
Null things that are disposed

Reply:
Yep, ?. then set them to null should fix that